File: LeetCode/周赛-243/5775. 准时抵达会议现场的最小跳过休息次数.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int minSkips(vector<int>& dist, int speed, int hoursBefore) {
        priority_queue<double> pq;
        int n = dist.size(), res=0;
        double cost=0, cost_sum=0;
        
        for (int i=0; i < n; i++) {
            cost = (double)dist[i] / speed;
            if (i != n-1) {
                pq.push(ceil(cost) - cost);
                cost_sum += ceil(cost);
            } else {
                cost_sum += cost;
            }
        }
        while (!pq.empty() && cost_sum > hoursBefore) {
            cout<<cost_sum<<','<<hoursBefore<<endl;
            cost_sum -= pq.top();
            pq.pop();
            res++;
        }
        if (cost_sum > hoursBefore) {
            return -1;
        }
        return res;
    }
};

File: LeetCode/竞赛/周赛/201-250/周赛-243/5775. 准时抵达会议现场的最小跳过休息次数.py
class Solution:
    def minSkips(self, dist: List[int], speed: int, hoursBefore: int) -> int:
        n = len(dist)
        res = 0
        pre_sum = [dist[0] for _ in range(n)]
        for i in range(1, n):
            pre_sum[i] = pre_sum[i-1] + dist[i]
        if pre_sum[-1] / speed > hoursBefore:
            return -1
        